src/resolve/resolved-conf.c

```diff
@@ -414,3 +414,61 @@ int manager_parse_config_file(Manager *m) {
         return 0;
 
 }
+
+int config_parse_refuse_record_types(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Manager *m = ASSERT_PTR(userdata);
+
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                data = mfree(data);
+                return 1;
+        }
+
+        r = free_and_strdup_warn(data, empty_to_null(rvalue));
+        if (r < 0)
+               return r;
+
+        char* record_type_string;
+        int refused_record_type;
+
+        /* Get int values of DNS type for example "AAAA" string to get int value and store in Set */
+        Set *refused_records = NULL;
```

Comment:
This is leaked on failure. Please use `_cleanup_set_free_` attribute.
BTW, I do not think this is necessary. See comment below.

Reply:
`_cleanup_set_free_` will cause a segfault where the data was freed before it can be used by `Manager`

Reply:
once you finished putting together the `Set*` object with all its contents, you should install it on the manager, and set it to NULL here, so that the `_cleanup_` handler is "deactivated", if you so will. This is typically done via a call to the `TAKE_PTR()` macro. That macro just shortens this: `p = q; q = NULL;` to `p = TAKE_PTR(q);` but does the same thing, i.e. assignment + invalidation of the old pointer